packages/@aws-cdk/aws-iotevents/lib/state.ts

```diff
@@ -53,13 +121,38 @@ export class State {
   public _onEnterEventsHaveAtLeastOneCondition(): boolean {
     return this.props.onEnter?.some(event => event.condition) ?? false;
   }
+
+  private toStateJson(): CfnDetectorModel.StateProperty {
+    const { stateName, onEnter } = this.props;
+    return {
+      stateName,
+      onInput: {
+        transitionEvents: toTransitionEventJson(this.transitionEvents),
+      },
+      onEnter: onEnter && { events: toEventJson(onEnter) },
+    };
+  }
 }
 
-function getEventJson(events: Event[]): CfnDetectorModel.EventProperty[] {
+function toEventJson(events: Event[]): CfnDetectorModel.EventProperty[] {
```

Comment:
Since `toTransitionEventJson()` is called first in the code, let's move it before this function.